src/System.Net.HttpListener/tests/HttpListenerAuthenticationTest.cs

```diff
@@ -384,22 +390,21 @@ public void Realm_SetDisposed_ThrowsObjectDisposedException()
 
         public async Task<HttpResponseMessage> AuthenticationFailure(HttpClient client, HttpStatusCode errorCode)
         {
+            client.Timeout = new TimeSpan(0, 0, 1);
             Task<HttpResponseMessage> clientTask = client.GetAsync(_factory.ListeningUrl);
 
             // The server task will hang forever if it is not cancelled.
             var tokenSource = new CancellationTokenSource();
             Task<HttpListenerContext> serverTask = Task.Run(() => _listener.GetContext(), tokenSource.Token);
-
+            
             // The client task should complete first - the server should send a 401 response.
             Task resultTask = await Task.WhenAny(clientTask, serverTask);
             tokenSource.Cancel();
             if (resultTask == serverTask)
             {
-                await serverTask;
+                await clientTask;
```

Comment:
Also, the change you made is very likely going to result in deadlock (without the timeout you added, which is problematic for other reasons).  If the server task completes first, that means the server failed for some reason, and the client will wait indefinitely (until it's timeout).  The change makes no sense to me.